ui/components-react/src/components-react/common/ImageRenderer.tsx

```diff
@@ -34,9 +34,33 @@ export class ImageRenderer {
     return (<img src={`data:image/${format};base64,${dataAsBase64}`} alt="" />);
   }
 
+  private isSvg(input: string): boolean {
+    const parser = new DOMParser();
+    const doc = parser.parseFromString(input, "application/xml");
+
+    const errorNode = doc.querySelector("parsererror");
+    if (errorNode) {
+      return false;
+    }
+
+    const rootNode = doc.documentElement.nodeName.toLowerCase();
+    return "svg" === rootNode;
+  }
+
+  private convertSvgToDataUri(svg: string) {
+    if (!this.isSvg(svg)) {
+      return "";
+    }
+
+    // eslint-disable-next-line deprecation/deprecation
+    const svgAsBase64 = btoa(svg);
+    return `data:image/svg+xml;base64,${svgAsBase64}`;
+  }
+
   /** Render svg string into JSX */
   private renderSvg(svg: string) {
-    const iconSpec = IconSpecUtilities.createWebComponentIconSpec(svg);
+    const svgAsDataUri = this.convertSvgToDataUri(svg.trim());
```

Comment:
>What do you mean by "if we shouldn't save the svgAsDataUri part"?

In this line, if we render the same input over and over again, the result will be the same but will always run the full conversion each time, even if the input are the same (except if the components using this are using React.memo or something like that)

So I was wondering if we shouldnt just return the result if the input was already calculated. But I'm not sure how to validate if this is indeed something that could cause performance problem, and I dont think we have any "UI performance" tests, or how we could do it.

Reply:
To resolve the potential performance issue we could introduce an in-memory cache, which would map the `svg` to `svgAsDataUri`. Do you think I should add it in this PR? This might increase the memory footprint.

Reply:
As this is a class with instances method, I would probably just initially keep the last value, not a full map.

Reply:
@raplemie I added a static map where key is the input `svg` string, and the value is the `svgAsDataUri`. Does it look okay?

Reply:
Looks good